insights/src/main/java/org/jboss/as/insights/extension/InsightsSubsystemAdd.java

```diff
@@ -0,0 +1,91 @@
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.dmr.ModelNode;
+import org.jboss.threads.JBossThreadFactory;
+
+import static java.security.AccessController.doPrivileged;
+import static org.jboss.as.insights.extension.InsightsSubsystemDefinition.ATTRIBUTES;
+import static org.jboss.as.insights.extension.InsightsSubsystemDefinition.INSIGHTS_RUNTIME_CAPABILITY;
+
+import org.jboss.as.controller.AttributeDefinition;
+
+/**
+ * Handler responsible for adding the subsystem resource to the model
+ *
+ * @author <a href="jkinlaw@redhat.com">Josh Kinlaw</a>
+ */
+class InsightsSubsystemAdd extends AbstractAddStepHandler {
+
+    static final InsightsSubsystemAdd INSTANCE = new InsightsSubsystemAdd();
+
+    private InsightsSubsystemAdd() {
+        super(INSIGHTS_RUNTIME_CAPABILITY, ATTRIBUTES);
+    }
+
+    @Override
+    protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+        boolean enabled = InsightsSubsystemDefinition.ENABLED.resolveModelAttribute(context, operation).asBoolean();
+        InsightsConfiguration config = new InsightsConfiguration(
+                getStringValue(InsightsSubsystemDefinition.INSIGHTSENDPOINT, context, operation),
+                getStringValue(InsightsSubsystemDefinition.SYSTEMENDPOINT, context, operation),
+                getStringValue(InsightsSubsystemDefinition.RHNUID, context, operation),
+                getStringValue(InsightsSubsystemDefinition.RHNPW, context, operation),
+                getStringValue(InsightsSubsystemDefinition.PROXYURL, context, operation),
+                InsightsSubsystemDefinition.PROXYPORT.resolveModelAttribute(context, operation).asInt(),
+                getStringValue(InsightsSubsystemDefinition.PROXYUSER, context, operation),
+                getStringValue(InsightsSubsystemDefinition.PROXYPASSWORD, context, operation),
+                getStringValue(InsightsSubsystemDefinition.URL, context, operation),
+                getStringValue(InsightsSubsystemDefinition.USERAGENT, context, operation),
+                InsightsSubsystemDefinition.SCHEDULE_INTERVAL.resolveModelAttribute(context, operation).asInt());
+        InsightsService.addService(context.getServiceTarget(), createUploaderExecutorService(), enabled, config);
+    }
+
+    private String getStringValue(AttributeDefinition def, OperationContext context, ModelNode operation) throws OperationFailedException {
+        ModelNode result = def.resolveModelAttribute(context, operation);
+        if(result.isDefined()) {
+            return result.asString();
+        }
+        return null;
+    }
+
+    @Override
+    protected boolean requiresRuntime(OperationContext context) {
+        return true;
```

Comment:
This should not override the superclass. The superclass will ensure this only adds the service on server processes, not on the Host Controllers. JDR itself is only added on servers, so there is no benefit adding this service on the Host Controller. And, if we did want this for the Host Controller process, there is additional work needed. If you leave this always returning true, on a Host Controller if there are 2 profiles that include this subsystem, they will both try to add the service and that will fail due to a duplicate service.